docs/aws/audit/cloudformationmonitoring/rules/enable_stack_notifications_remediation.mdx

```diff
@@ -1,6 +1,248 @@
+       else:
+           print("Notifications are not enabled for the stack.")
+   else:
+       print("Stack not found.")
+   ```
+
+By following these steps, you can ensure that your AWS CloudFormation stacks have notifications enabled, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the CloudFormation service. You can find this by typing 'CloudFormation' into the search bar at the top of the console, then selecting 'CloudFormation' from the dropdown menu.
+3. In the CloudFormation console, you will see a list of your stacks. Click on the name of the stack you want to check.
+4. In the stack details page, click on the 'Settings' tab. Under the 'Notification ARNs' section, you can see if any Amazon SNS topics are set up to receive notifications about the stack's events. If this section is empty, then notifications are not enabled for this stack.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   ```
+   pip install awscli
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFormation stacks: Use the following command to list all the CloudFormation stacks in your AWS account:
+
+   ```
+   aws cloudformation list-stacks --query 'StackSummaries[*].[StackName]' --output text
+   ```
+   This command will return a list of all stack names.
+
+3. Check if notifications are enabled: For each stack, you can check if notifications are enabled by describing the stack and looking for the "NotificationARNs" attribute. Use the following command:
+
+   ```
+   aws cloudformation describe-stacks --stack-name <StackName> --query 'Stacks[*].[NotificationARNs]' --output text
+   ```
+   Replace `<StackName>` with the name of the stack you want to check. If the output is empty, it means that notifications are not enabled for this stack.
+
+4. Repeat step 3 for all stacks: You need to repeat the previous step for all the stacks listed in step 2. This way, you can check if notifications are enabled for all your CloudFormation stacks.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>